src/policy_metadata.rs

```diff
@@ -0,0 +1,388 @@
+        return Err(ValidationError::new(
+            "No other elements can be defined when '*' is used",
+        ));
+    }
+    Ok(())
+}
+
+fn validate_resources(data: &[String]) -> Result<(), ValidationError> {
+    // This method is a transposition of the check done by Kubernetes
+    // see https://github.com/kubernetes/kubernetes/blob/09268c16853b233ebaedcd6a877eac23690b5190/pkg/apis/admissionregistration/validation/validation.go#L44
+
+    // */x
+    let mut resources_with_wildcard_subresources: HashSet<String> = HashSet::new();
+    // x/*
+    let mut subresources_with_wildcard_resource: HashSet<String> = HashSet::new();
+    // */*
+    let mut has_double_wildcard = false;
+    // *
+    let mut has_single_wildcard = false;
+    // x
+    let mut has_resource_without_subresource = false;
+
+    for resource in data.iter() {
+        if resource.is_empty() {
+            return Err(ValidationError::new("empty resource is not allowed"));
+        }
+        match resource.as_str() {
+            "*/*" => has_double_wildcard = true,
+            "*" => has_single_wildcard = true,
+            _ => {}
+        };
+
+        let parts: Vec<&str> = resource.splitn(2, '/').collect();
+        if parts.len() == 1 {
+            has_resource_without_subresource = resource.as_str() != "*";
+            continue;
+        }
+        let res = parts[0];
+        let sub = parts[1];
+
+        if resources_with_wildcard_subresources.contains(res) {
+            let msg = format!("if '{}/*' is present, must not specify {}", resource, res);
+            return Err(ValidationError::new(Box::leak(msg.into_boxed_str())));
+        }
+        if subresources_with_wildcard_resource.contains(sub) {
+            let msg = format!("if '*/{}' is present, must not specify {}", sub, resource);
+            return Err(ValidationError::new(Box::leak(msg.into_boxed_str())));
+        }
+        if sub == "*" {
+            resources_with_wildcard_subresources.insert(String::from(res));
+        }
+        if res == "*" {
+            subresources_with_wildcard_resource.insert(String::from(sub));
+        }
+    }
+    if data.len() > 1 && has_double_wildcard {
+        return Err(ValidationError::new(
+            "if '*/*' is present, must not specify other resources",
+        ));
+    }
+    if has_single_wildcard && has_resource_without_subresource {
+        return Err(ValidationError::new(
+            "if '*' is present, must not specify other resources without subresources",
+        ));
+    }
+
+    Ok(())
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Validate)]
+#[serde(rename_all = "camelCase")]
+pub struct Metadata {
+    #[serde(default = "protocol_version_unspecified")]
+    pub protocol_version: ProtocolVersion,
+    #[validate]
+    pub rules: Vec<Rule>,
+    pub labels: HashMap<String, String>,
+}
+
+fn protocol_version_unspecified() -> ProtocolVersion {
+    ProtocolVersion::V0
```

Comment:
I think we should have `v0` and something that reflects `unknown` when unset. And if cannot be unmarshaled. When defaulting here, I'd rather use `unknown` rather than `v0` if we have the latest. Point is: if I'm developing, and I want to avoid the `annotate` hassle, everything will assume "latest".